src/NuGet.Core/NuGet.Packaging/Core/FrameworkNameValidatorUtility.cs

```diff
@@ -34,6 +34,51 @@ internal static bool IsValidFrameworkName(string path)
             return fx != null && fx.Framework != NuGetFramework.UnsupportedFramework.Framework;
         }
 
+        internal static bool IsDottedFrameworkVersion(string path)
+        {
+            foreach (string knownFolder in PackagingConstants.Folders.Known)
+            {
+                string folderPrefix = knownFolder + System.IO.Path.DirectorySeparatorChar;
+                if (path.Length > folderPrefix.Length &&
+                    path.StartsWith(folderPrefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    string frameworkPart = path.Substring(folderPrefix.Length);
+
+                    try
+                    {
+                        string effectivePath = null;
+                        NuGetFramework fw = FrameworkNameUtility.ParseNuGetFrameworkFolderName(
+                            frameworkPart,
+                            strictParsing: knownFolder == PackagingConstants.Folders.Lib,
+                            effectivePath: out effectivePath);
+
+                        string targetFrameworkString = Path.GetDirectoryName(path).Split(Path.DirectorySeparatorChar).ElementAt(1);
+
+                        var isNet5EraTfm = fw.Version.Major >= 5 && StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, fw.Framework);
```

Comment:
Can we create an utility method to test .NET 5 era condition? I have seen this in multiple parts of the codebase

```c#
public static bool isNet5Era(Version targetFrameworkVersion, string targetFrameworkIdentifier);
```

See also:

https://github.com/NuGet/NuGet.Client/blob/0e5413751caebf168e7d363c88b9c22a3560fe60/src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs#L94

https://github.com/NuGet/NuGet.Client/blob/0e5413751caebf168e7d363c88b9c22a3560fe60/src/NuGet.Core/NuGet.Frameworks/NuGetFrameworkFactory.cs#L139-L140

Reply:
This came up in review (I originally had this property set in NuGetFramework), but the feedback that I got pretty consistently is that people didn't want to have this as part of that class. /cc @zivkan @nkolev92

Reply:
I don't like the name `IsNet5Era`. It's acceptable for internal use, but public APIs should be 1. as obvious/easy to understand as possible, without reading docs 2. try to be "future proof".

Will a `IsNet5Era` API be easy to understand in 5 or 10 years from now when everyone is using .NET 10, .NET 15? Even now, will someone browsing our public APIs via IntelliSense be able to easily guess what the API means, and when they should use it?

But given we do the check in several assemblies, there's a good chance that some customers will want to as well. Maybe `DoesTargetFrameworkUsePlatform(...)`?  `IsPlatformUsed(...)`? `UsesPlatform(...)`? `UsesPlatformMoniker(..)`?

Also, given the TFM format is `{tfi},Version=v{tfv}`, I'd suggest that the parameters should be ordered `(string identifier, Version version)`, regardless of what order they're used in the method body.